src/main/java/me/botsko/prism/appliers/Preview.java

```diff
@@ -24,486 +16,405 @@
+        if (player == null)
+            return;
+        if (!blockStateChanges.isEmpty()) {
+
+            // pull all players that are part of this preview
+            final List<CommandSender> previewPlayers = parameters.getSharedPlayers();
+            previewPlayers.add(player);
+
+            for (final BlockStateChange u : blockStateChanges) {
+                Location loc = u.getOriginalBlock().getLocation();
+                BlockData data = u.getOriginalBlock().getBlockData();
+
+                for (final CommandSender sharedPlayer : previewPlayers) {
+                    if (sharedPlayer instanceof Player)
+                        EntityUtils.sendBlockChange((Player) sharedPlayer, loc, data);
+                }
+            }
+        }
+        sender.sendMessage(
+                Prism.messenger.playerHeaderMsg("Preview canceled." + ChatColor.GRAY + " Please come again!"));
+    }
+
+    /**
+     *
+     */
+    @Override
+    public void apply_preview() {
+        if (player == null)
+            return;
+        sender.sendMessage(Prism.messenger.playerHeaderMsg("Applying rollback from preview..."));
+        setIsPreview(false);
+        changesAppliedCount = 0;
+        skippedBlockCount = 0;
+        changesPlannedCount = 0;
+        apply();
+    }
+
+    /**
+     *
+     */
+    @Override
+    public void preview() {
+    }
+
+    @Override
+    public void apply() {
+
+        if (!worldChangeQueue.isEmpty()) {
+
+            if (!isPreview && player != null) {
+
+                Wand oldwand = null;
+                if (Prism.playersWithActiveTools.containsKey(player.getName())) {
+                    // Pull the wand in use
+                    oldwand = Prism.playersWithActiveTools.get(player.getName());
+                }
+
+                boolean show_nearby = true;
+                if (oldwand instanceof RollbackWand) {
+                    show_nearby = false;
+                }
+                if (show_nearby) {
+                    // Inform nearby players
+                    plugin.notifyNearby(player, parameters.getRadius(), player.getDisplayName() + " is performing a "
+                            + processType.name().toLowerCase() + " near you.");
+                    // Inform staff
+                    if (plugin.getConfig().getBoolean("prism.alerts.alert-staff-to-applied-process")) {
+                        final String cmd = parameters.getOriginalCommand();
+                        if (cmd != null) {
+                            plugin.alertPlayers(player, ChatColor.WHITE + processType.name().toLowerCase() + " by "
+                                    + player.getDisplayName() + ChatColor.GRAY + parameters.getOriginalCommand());
+                        }
+                    }
+                }
+            }
+
+            // Offload the work of world changes
+            // to a scheduled sync task
+            processWorldChanges();
+        }
+    }
+
+    private void processWorldChanges() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'processWorldChanges()' has an NPath complexity of 74912](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)